### PR TITLE
ENH/FIX add format kwarg to TimeFrequencyScattering torch and tf frontends

### PR DESCRIPTION
**Pull request by cyrusvahidi**
I have added the `format` kwarg for `TimeFrequencyScattering` torch and tf frontends. Upon testing, I have noticed a `RuntimeError` for `format="time"` in tensorflow

### CODE REVIEW

**Review by lostanlen — APPROVED** [review]

File: kymatio/scattering1d/frontend/tensorflow_frontend.py [hunk 1]
```diff
@@ -20,22 +31,46 @@ def __init__(self, J, shape, Q=1, T=None, max_order=2, oversampling=0,
...
+        shape,
+        Q,
+        T=None,
+        oversampling=0,
+        Q_fr=1,
+        F=None,
+        oversampling_fr=0,
+        out_type="array",
+        backend="tensorflow",
+        format="joint",
```
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/tensorflow_frontend.py:50]: what does this do

Comment (MuawizChaudhary) [reply kymatio/scattering1d/frontend/tensorflow_frontend.py:50]: what does the format kwarg do

Comment (cyrusvahidi) [reply kymatio/scattering1d/frontend/tensorflow_frontend.py:50]: the format kwarg organizes the output shape in 2D (`"time"`) or 3D (`"joint"`). in 2D, it returns `(path, time`) while 3D returns `(path, freq, time)`. the latter allows 2D convolution

Comment (cyrusvahidi) [reply kymatio/scattering1d/frontend/tensorflow_frontend.py:50]: previously the kwarg was implemented but missing for the `torch` frontend. this PR resolves that

Comment (MuawizChaudhary) [reply kymatio/scattering1d/frontend/tensorflow_frontend.py:50]: can you link me to the code where it was implemented to do that stuff you mention?

Comment (cyrusvahidi) [reply kymatio/scattering1d/frontend/tensorflow_frontend.py:50]: https://github.com/cyrusvahidi/kymatio/blob/6b35444be93e7cd9d19d8c5db061a5240d6bf239/kymatio/scattering1d/core/timefrequency_scattering.py#L409

Comment (MuawizChaudhary) [reply kymatio/scattering1d/frontend/tensorflow_frontend.py:50]: Thank you.

**Pull request merged**